dripline/extensions/thermo_fisher_endpoint.py

```diff
@@ -0,0 +1,82 @@
+from dripline.core import Entity, calibrate, ThrowReply
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = []
+__all__.append("ThermoFisherGetHexEntity")
+__all__.append("ThermoFisherGetEntity")
+
+
+class ThermoFisherGetHexEntity(Entity):
+    '''
+    A simple endpoint which stores a value but returns that value + a random offset
+    '''
+
+    def __init__(self, 
+                 get_str=None,
+                 **kwargs):
+        '''
+        Args:
+            cmd_str: hexstring of the command, e.g. 20 
+        '''
+        if get_str is None:
+            raise ValueError('<base_str is required to __init__ ThermoFisherGetEntity instance')
+        else:
+            self.cmd_str = str(get_str).zfill(2)
+        Entity.__init__(self, **kwargs)
+
+    @calibrate()
+    def on_get(self):
+        # setup cmd here
+        to_send = [self.cmd_str]
+        logger.debug(f'Send cmd in hexstr: {to_send[0]}')
+        result = self.service.send_to_device(to_send)
+        logger.debug(f'raw result is: {result}')
+
+        return result
+
+    def on_set(self, value):       
+        raise ThrowReply('message_error_invalid_method', f"endpoint '{self.name}' does not support set")
+        to_send = [cmd]
+        result = self.service.send_to_device(to_send)
+        logger.debug(f'raw result is: {result}')
+        # do something here
+        return result
+
+
+class ThermoFisherGetEntity(ThermoFisherGetHexEntity):
```

Comment:
without seeing how these will be used, this name feels like it could be more descriptive - seeing a "GetEntity" and a "GetHexEntity" I might guess that this was more generic, rather than applying an additional correction to treat floats properly

Reply:
What about GetNumericEntity ? That would also solve the question about value + unit string or just value?